File: data_structures/hash_table/hash_table.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        // Creating a hash table for storing phone numbers
        Dictionary<string, string> phoneBook = new Dictionary<string, string>();

        // Adding contacts to the hash table
        phoneBook["Alice"] = "+123456789";
        phoneBook["Bob"] = "+987654321";
        phoneBook["Charlie"] = "+111223344";

        // Getting phone numbers by name
        string aliceNumber = phoneBook.TryGetValue("Alice", out var alice) ? alice : "Number not found";
        string bobNumber = phoneBook.TryGetValue("Bob", out var bob) ? bob : "Number not found";
        string daveNumber = phoneBook.TryGetValue("Dave", out var dave) ? dave : "Number not found";

        // Displaying information about contacts
        Console.WriteLine($"Alice's number: {aliceNumber}");
        Console.WriteLine($"Bob's number: {bobNumber}");
        Console.WriteLine($"Dave's number: {daveNumber}");
    }
}

File: 1 data_structures/hash_table/hash_table.py
# Створення хеш-таблиці для зберігання телефонних номерів
phone_book = {}

# Додавання контактів до хеш-таблиці
phone_book["Alice"] = "+123456789"
phone_book["Bob"] = "+987654321"
phone_book["Charlie"] = "+111223344"

# Отримання номера телефону за іменем
alice_number = phone_book.get("Alice", "Number not found")
bob_number = phone_book.get("Bob", "Number not found")
dave_number = phone_book.get("Dave", "Number not found")

# Виведення інформації про контакти
print(f"Alice's number: {alice_number}")
print(f"Bob's number: {bob_number}")
print(f"Dave's number: {dave_number}")
